### DISCUSSION TITLE
Apply the same transformation to field + all fields that reference it

### DISCUSSION THREAD

**Discussion by janmeier** [GreenmaskIO/greenmask#158] [Ideas]
So, we have a table named `coach`, and we want to anonymize the `id` column, so that it's not possible to take an id from our staging environment and search for it in our production environment, to de-anonymize the data.

However, if we do that, we would also need to apply the same transformation to all the fields that reference `coach.id`.

Of course we could do that manually, but the table is referenced in many places, so it would clutter the configuration file, and we would have to remember to update the transformations every time we make a new table that references coach.

One very simple way to solve it would be to have "global" transformers defined, so that every time we encounter a field name `coachId`, it gets transformed with `RandomUUID`, using the `hash` engine. That would solve the issue for our usecase.

However, since it looks like you are already working on analyzing foreign keys in connection with subsetting, it might make sense to do this automagically if you have all the required information available already? Maybe we could add a setting on the transformation of the `coach.id` column called something like `transformReferences: true`, and that would take care of applying the transformation to all referencing columns?

**Comment by wwoytenko** [comment]
Yeah, that's a good point. Since we've recently almost completed the database subset system and already have the reference graph defined for the tables, we can implement this feature. The feature request #153, which was resolved by MR #157, makes it possible to implement this soon. I'll aim to add this feature in version 0.2b2 by the end of August. I will create an issue with the feature request next week.

**Comment by wwoytenko** [comment]
I've just created the feature request here #182

**Comment by wwoytenko** [comment]
Hi @janmeier! The major release is scheduled for October 10th! 
However, task [#182](https://github.com/GreenmaskIO/greenmask/issues/182) will not be included as I need to conduct some additional testing. It will, however, be released by the end of October. I’ll keep you updated!

**Comment by wwoytenko** [comment]
@janmeier it has been implemented and merged in #229. The release will be published soon. I will keep you posted.

**Comment by wwoytenko** [comment]
Published in [v0.2.1](https://github.com/GreenmaskIO/greenmask/releases/tag/v0.2.1)

I would appreciate any feedback. Thanks!

**Reply by wwoytenko** [reply]
Related doc is here
https://docs.greenmask.io/latest/built_in_transformers/transformation_inheritance/
